### PR TITLE
DTFS-3846—fe industry

### PR DESCRIPTION
**Pull request by monkemedia**

### CODE REVIEW

File: gef-ui/server/controllers/about-exporter/index.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
 import * as api from '../../services/api';
 import { validationErrorHandler, isTrueSet } from '../../utils/helpers';
 
+const mappedIndustries = (industries, selectedIndustry) => {
+  if (!industries) { return null; }
+  return industries.map((industry) => ({
+    value: JSON.stringify(industry),
+    html: `${industry.name}<br />${industry.class.name}`,
+    checked: selectedIndustry ? JSON.stringify(industry) === selectedIndustry : null,
+  }));
```
Comment (ttbarnes) [inline gef-ui/server/controllers/about-exporter/index.js:10]: i'd have any html in the template itself, not a blocker to merge

File: gef-ui/server/controllers/about-exporter/index.js [hunk 2]
```diff
@@ -29,47 +40,58 @@ const validateAboutExporter = async (req, res) => {
...
+        aboutExporterErrors.push({
+          errRef: 'probabilityOfDefault',
+          errMsg: 'Enter the probability of default',
+        });
+      }
+
+      if (!body.isFinanceIncreasing) {
+        aboutExporterErrors.push({
+          errRef: 'isFinanceIncreasing',
+          errMsg: 'Enter the elect whether your financing to the exporter is increasing as a result of this new GEF facility ',
```
Comment (ttbarnes) [inline gef-ui/server/controllers/about-exporter/index.js:73]: Should this be "Slect whether ..." ?

File: gef-ui/server/utils/helpers.js [hunk 3]
```diff
@@ -63,19 +64,15 @@ const mapSummaryList = (data, itemsToShow) => {
       return { html: '<span class="has-text-danger" data-cy="required">Required</span>' };
     }
 
-    if (val === null) {
+    /* Clean-Deep removes any properties with Null value from an Object. Therefore if all
+    properties are Null, this leaves us with an Empty Object. isEmpty checks to see if the
+    Object is empty or not. */
+    if (val === null || _isEmpty(cleanDeep(val))) {
```
Comment (ttbarnes) [inline gef-ui/server/utils/helpers.js:70]: Could just have a nunjuck filter to display dash if there is no value?
```<p>{{ someValue | dashIfEmpty }}</p>```

https://github.com/foundry4/dtfs2/blob/master/gef-ui/server/nunjucks-configuration/filter-dashIfEmpty.js

I think either works though. Probably more of a personal preference. What do you think?

File: gef-ui/server/utils/helpers.js [hunk 4]
```diff
@@ -123,7 +119,10 @@ const mapSummaryList = (data, itemsToShow) => {
         items: [
           ...(href ? [{
             href,
-            text: `${value ? 'Change' : 'Add'}`,
+            /* Clean-Deep removes any properties with Null value from an Object. Therefore if all
+            properties are Null, this leaves us with an Empty Object. isEmpty checks to see if the
+            Object is empty or not. */
+            text: `${!_isEmpty(cleanDeep(value)) ? 'Change' : 'Add'}`,
```
Comment (ttbarnes) [inline gef-ui/server/utils/helpers.js:125]: probably worth putting into it's on function as it's duplicated, maybe something like

```
const isEmpty = (value) =>
  _isEmpty(cleanDeep(value)
```

**Review by ttbarnes — COMMENTED** [review]
LGTM, couple of minor comments

File: gef-ui/server/controllers/about-exporter/index.js (hunk 1 above)
Comment (monkemedia) [reply gef-ui/server/controllers/about-exporter/index.js:10]: The radio buttons are dynamic, so wont be able to hard code them in. Is there a way to use a for loop for items in the njk file?

File: gef-ui/server/utils/helpers.js (hunk 3 above)
Comment (monkemedia) [reply gef-ui/server/utils/helpers.js:70]: The whole summary list is generated via JS, so there is html in the template.

File: gef-ui/server/controllers/about-exporter/index.js (hunk 1 above)
Comment (ttbarnes) [reply gef-ui/server/controllers/about-exporter/index.js:10]:
```
{% for industry in industries %}
  <p>{{ industry.name }}</p>
{% endfor %}
```

File: gef-ui/server/utils/helpers.js (hunk 4 above)
Comment (monkemedia) [reply gef-ui/server/utils/helpers.js]: Done. Also added unit tests for it too

File: gef-ui/server/controllers/about-exporter/index.js (hunk 1 above)
Comment (monkemedia) [reply gef-ui/server/controllers/about-exporter/index.js:10]: so this is my template code for the radio buttons:

`{{ govukRadios({
     idPrefix: "industry",
     name: "selectedIndustry",
     items: industries, // Mapped in the controller
     attributes: {
       'data-cy': 'industries'
     }
}) }}`

So are you saying to wrap this macro into a for loop?

Comment (ttbarnes) [reply gef-ui/server/controllers/about-exporter/index.js:10]: Sorry, not familiar with design /  didn't realise there were radios involved. I think what you have is ok - I don't think it's possible to populate the govukRadios items attribute dynamically, I think, can only pass in an array and that's it 👍

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**